packages/react-core/src/components/Tooltip/Tooltip.tsx

```diff
@@ -50,150 +49,192 @@ export interface TooltipProps {
+    console.warn(
+      'The Tooltip boundary prop has been deprecated. If you want to constrain the popper to a specific element use the appendTo prop instead.'
+    );
+  }
+  // could make this a prop in the future (true | false | 'toggle')
+  const hideOnClick = true;
+  const triggerOnMouseenter = trigger.includes('mouseenter');
+  const triggerOnFocus = trigger.includes('focus');
+  const triggerOnClick = trigger.includes('click');
+  const triggerManually = trigger === 'manual';
+  const [visible, setVisible] = React.useState(false);
+  const [opacity, setOpacity] = React.useState(0);
+  const transitionTimerRef = React.useRef(null);
+  const showTimerRef = React.useRef(null);
+  const hideTimerRef = React.useRef(null);
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (!triggerManually) {
+      if (event.keyCode === KEY_CODES.ESCAPE_KEY && visible) {
+        hide();
+      } else if (event.keyCode === KEY_CODES.ENTER && !visible) {
+        show();
+      }
+    }
   };
-
-  handleEscKeyClick = (event: KeyboardEvent) => {
-    if (event.keyCode === KEY_CODES.ESCAPE_KEY && this.tip.state.isVisible) {
-      this.tip.hide();
+  React.useEffect(() => {
+    if (triggerManually) {
+      if (isVisible) {
+        show();
+      } else {
+        hide();
+      }
     }
+  }, [isVisible, triggerManually]);
+  const show = () => {
+    if (transitionTimerRef.current) {
+      clearTimeout(transitionTimerRef.current);
+    }
+    if (hideTimerRef.current) {
+      clearTimeout(hideTimerRef.current);
+    }
+    showTimerRef.current = setTimeout(() => {
+      setVisible(true);
+      setOpacity(1);
+    }, entryDelay);
   };
+  const hide = () => {
+    if (showTimerRef.current) {
+      clearTimeout(showTimerRef.current);
+    }
+    hideTimerRef.current = setTimeout(() => {
+      setOpacity(0);
+      transitionTimerRef.current = setTimeout(() => setVisible(false), animationDuration);
+    }, exitDelay);
+  };
+  const positionModifiers = {
+    top: styles.modifiers.top,
+    bottom: styles.modifiers.bottom,
+    left: styles.modifiers.left,
+    right: styles.modifiers.right
+  };
+  const hasCustomMaxWidth = maxWidth !== tooltipMaxWidth.value;
+  const content = (
+    <div
+      className={css(styles.tooltip, className)}
+      role="tooltip"
+      id={id}
+      style={{
+        maxWidth: hasCustomMaxWidth ? maxWidth : null,
+        opacity,
+        transition: getOpacityTransition(animationDuration)
+      }}
+      {...rest}
+    >
+      <TooltipArrow />
+      <TooltipContent isLeftAligned={isContentLeftAligned}>{bodyContent}</TooltipContent>
+    </div>
+  );
 
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleEscKeyClick, false);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleEscKeyClick, false);
-  }
+  const onDocumentClick = (event: MouseEvent, triggerElement: HTMLElement) => {
+    // event.currentTarget = document
```

Comment:
can we remove these commented lines?

Reply:
it's a helpful comment to know what is on the event